internal/elasticsearch/security/user/update.go

```diff
@@ -0,0 +1,129 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/elasticsearch"
+	"github.com/elastic/terraform-provider-elasticstack/internal/diagutil"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (r *userResource) update(ctx context.Context, plan tfsdk.Plan, state *tfsdk.State) diag.Diagnostics {
+	var planData UserData
+	var diags diag.Diagnostics
+	diags.Append(plan.Get(ctx, &planData)...)
+	if diags.HasError() {
+		return diags
+	}
+
+	// Check if we have existing state (this is an update, not a create)
+	hasState := false
+	var stateData UserData
+	if state != nil && !state.Raw.IsNull() {
+		hasState = true
+		diags.Append(state.Get(ctx, &stateData)...)
+		if diags.HasError() {
+			return diags
+		}
+	}
+
+	usernameId := planData.Username.ValueString()
+	id, sdkDiags := r.client.ID(ctx, usernameId)
+	diags.Append(diagutil.FrameworkDiagsFromSDK(sdkDiags)...)
+	if diags.HasError() {
+		return diags
+	}
+
+	client, connDiags := clients.MaybeNewApiClientFromFrameworkResource(ctx, planData.ElasticsearchConnection, r.client)
+	diags.Append(connDiags...)
+	if diags.HasError() {
+		return diags
+	}
+
+	var user models.User
+	user.Username = usernameId
+
+	// Handle password fields - only set password if it's in the plan AND (it's a create OR it has changed from state)
+	// Priority: password_wo > password > password_hash
+	if utils.IsKnown(planData.PasswordWo) && (!hasState || !planData.PasswordWoVersion.Equal(stateData.PasswordWoVersion)) {
```

Comment:
The Terraform [docs](https://developer.hashicorp.com/terraform/plugin/framework/resources/write-only-arguments#retrieving-write-only-values) state:

> Write-only argument values should be retrieved from the configuration instead of the plan. 

Update this function to read the `password_wo` attribute from config rather than the plan. You'll need to:
* Pass the config through as an argument alongside plan/state.
* Use `config.GetAttribute` to just read the `password_wo` attribute from config.